config/src/main/java/tech/pegasys/pantheon/config/GenesisConfigOptions.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.config;
+
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+
+import io.vertx.core.json.JsonObject;
+
+public class GenesisConfigOptions {
+
+  private static final String IBFT_CONFIG_KEY = "ibft";
+  private static final String CLIQUE_CONFIG_KEY = "clique";
+  private final JsonObject configRoot;
+
+  private GenesisConfigOptions(final JsonObject configRoot) {
+    this.configRoot = configRoot != null ? configRoot : new JsonObject();
+  }
+
+  public static GenesisConfigOptions fromGenesisConfig(final String genesisConfig) {
+    return fromGenesisConfig(new JsonObject(genesisConfig));
+  }
+
+  public static GenesisConfigOptions fromGenesisConfig(final JsonObject genesisConfig) {
+    return new GenesisConfigOptions(genesisConfig.getJsonObject("config"));
+  }
+
+  public boolean isEthHash() {
+    return configRoot.containsKey("ethash");
```

Comment:
is there a reason this doesn't do the same as IBFT/clique with regard a class constant?

Reply:
That string was only referenced once so locality of reference was very useful, but will move it for consistency since it is very much like the IBFT and Clique keys so should follow the same pattern.